### PR TITLE
Add release GHA

### PR DESCRIPTION
**Pull request by ejguan**
Copy of #220 in order to access env secrets

### Changes
- Update `setup.py` to generate right version for different types of releases (nightly, rc, official release)
- Add GHA workflow to enable to release uploading for testing and nightly
- Currently using manually dispatch to trigger this upload for now. And, manual trigger can only be carried out when this PR is landed.  Please check this test run to validate the workflow: https://github.com/pytorch/data/actions/runs/1843730857

### CODE REVIEW

File: .github/workflows/release.yml
Comment (ejguan) [inline .github/workflows/release.yml:63]: These env variables are fake ones for now.

Comment (ejguan) [reply .github/workflows/release.yml:63]: The CI Error also came from such fake secrets variables.

File: .github/workflows/release.yml
Comment (ejguan) [inline .github/workflows/release.yml:15]: Will delete it before landing.

**Review by VitalyFedyunin — APPROVED** [review]

**Comment by VitalyFedyunin** [comment]
Can you please put short doc how to manually trigger release jobs.

File: .github/workflows/release.yml
Comment (ejguan) [inline .github/workflows/release.yml:9]: Adding notes here. @VitalyFedyunin 
Manually trigger needs to wait until this PR landed.

**Pull request closed without merging**
